src/notifications/endpoints/components/EndpointCardMenu.tsx

```diff
@@ -1,38 +1,84 @@
 // Libraries
-import React, {FC} from 'react'
+import React, {FC, useRef} from 'react'
 
 // Components
-import {Context, IconFont} from 'src/clockface'
-import {ComponentColor} from '@influxdata/clockface'
+import {
+  Popover,
+  List,
+  ConfirmationButton,
+  ButtonShape,
+  ComponentColor,
+  FlexBox,
+  ComponentSize,
+  IconFont,
+  SquareButton,
+  SquareButtonRef,
+  Appearance,
+} from '@influxdata/clockface'
 
 interface Props {
   onDelete: () => void
   onClone: () => void
   onView: () => void
 }
 
-const EndpointCardContext: FC<Props> = ({onDelete, onView}) => {
+const EndpointCardContext: FC<Props> = ({onDelete, onClone, onView}) => {
+  const triggerRef = useRef<SquareButtonRef>(null)
+
+  const handleCloneAndHide = (onHide: () => void) => (): void => {
+    onHide()
+    onClone()
+  }
+
+  const handleViewAndHide = (onHide: () => void) => (): void => {
+    onHide()
+    onView()
+  }
+
   return (
-    <Context>
-      <Context.Menu icon={IconFont.EyeOpen} testID="context-history-menu">
-        <Context.Item
-          label="View History"
-          action={onView}
-          testID="context-history-task"
-        />
-      </Context.Menu>
-      <Context.Menu
+    <FlexBox margin={ComponentSize.Small}>
+      <SquareButton
+        icon={IconFont.CogThick}
+        size={ComponentSize.ExtraSmall}
+        ref={triggerRef}
+        testID="context-endpoint-options"
+      />
+      <Popover
+        triggerRef={triggerRef}
+        appearance={Appearance.Outline}
+        enableDefaultStyles={false}
+        contents={onHide => (
+          <List style={{width: '122px'}}>
+            <List.Item
+              size={ComponentSize.ExtraSmall}
+              testID="context-endpoint-history"
+              onClick={handleViewAndHide(onHide)}
+            >
+              <List.Icon glyph={IconFont.EyeOpen} />
+              View History
+            </List.Item>
+            <List.Item
+              size={ComponentSize.ExtraSmall}
+              testID="context-clone-endpoint"
+              onClick={handleCloneAndHide(onHide)}
+            >
+              <List.Icon glyph={IconFont.Duplicate} />
+              Clone
+            </List.Item>
+          </List>
+        )}
+      />
```

Comment:
this `View History` and `Clone` behavior seems similar to the other card. is there enough repeated commonality to consider this a pattern?

Reply:
Yeah it's def a pattern. Not all resources (currently) have a clone button, but it's pretty common. I'd love to standardize it in the code so it's not up to each resource list to remember including it